controllers/UserController.php

```diff
@@ -142,16 +152,150 @@ public function actionProfile()
+    {
+        if (!Yii::$app->request->isPost) {
+            return $this->render('fields/change-username', ['usernameModel' => $this->user]);
+        }
+
+        $this->user->load(Yii::$app->request->post());
+
+        if ($this->user->validate() && $this->user->save()) {
+            Yii::$app->session->setFlash('success', 'Username changed.');
+            return $this->redirect('/account');
+        } else {
+            return $this->render('fields/change-username', ['usernameModel' => $this->user]);
+        }
+    }
+
+    public function actionChangeName()
+    {
+        if (!Yii::$app->request->isPost) {
+            return $this->render('fields/change-name', ['nameModel' => $this->user]);
+        }
+
+        $this->user->load(Yii::$app->request->post());
+
+        if ($this->user->validate() && $this->user->save()) {
+            Yii::$app->session->setFlash('success', 'Name changed.');
+            return $this->redirect('/account');
+        } else {
+            return $this->render('fields/change-name', ['nameModel' => $this->user]);
+        }
+    }
+
+    public function actionChangeBirthday()
+    {
+
+        if (!Yii::$app->request->isPost) {
+            return $this->render('fields/change-birthday', ['birthdayModel' => $this->user]);
+        }
+
+        $this->user->load(Yii::$app->request->post());
+        $this->user->birthday = Yii::$app->formatter->asDate($this->user->birthday);
+        if ($this->user->validate() && $this->user->save()) {
+            Yii::$app->session->setFlash('success', 'Birthday changed.');
+            return $this->redirect('/account');
+        } else {
+            return $this->render('fields/change-birthday', ['birthdayModel' => $this->user]);
+        }
+    }
+
+    public function actionChangeGender()
+    {
+        if (!Yii::$app->request->isPost) {
+            $genders = Gender::find()->select(['name', 'id'])->indexBy('id')->asArray()->column();
+            return $this->render('fields/change-gender', ['genderModel' => $this->user, 'genders' => $genders]);
+        }
+
+        $postData = Yii::$app->request->post('User');
+        $gender_id = $postData['gender'];
+        $this->user->gender_id = $gender_id;
+
+        if ($this->user->validate() && $this->user->save()) {
+            Yii::$app->session->setFlash('success', 'Gender changed.');
+            return $this->redirect('/account');
+        } else {
+            $genders = Gender::find()->select(['name', 'id'])->indexBy('id')->asArray()->column();
+            return $this->render('fields/change-gender', ['genderModel' => $this->user, 'genders' => $genders]);
+        }
+    }
+
+    public function actionChangeTimezone()
+    {
+        if (!Yii::$app->request->isPost) {
+            return $this->render('fields/change-timezone', ['timezoneModel' => $this->user]);
+        }
+
+        $this->user->load(Yii::$app->request->post());
+
+        if ($this->user->validate() && $this->user->save()) {
```

Comment:
There is no need to call `ActiveRecord::validate` method because it is called in `ActiveRecord::save` method.